File: js(ts)-solutions/1261FindElementsInAContaminatedBinaryTree.ts

// Definition for a binary tree node.
class TreeNode {
    val: number
    left: TreeNode | null
    right: TreeNode | null
    constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
        this.val = (val === undefined ? 0 : val)
        this.left = (left === undefined ? null : left)
        this.right = (right === undefined ? null : right)
    }
}

class FindElements {
    allNodes: Set<number>;
    constructor(root: TreeNode) {
        this.allNodes = new Set();
        const traverse = (root: TreeNode | null, currVal: number) => {
            if (root === null) return

            this.allNodes.add(currVal);
            traverse(root.left, currVal * 2 + 1);
            traverse(root.right, currVal * 2 + 2);
        }

        root.val = 0;
        traverse(root, 0)
    }

    find(target: number): boolean {
        return this.allNodes.has(target);
    }
}

/**
 * Your FindElements object will be instantiated and called as such:
 * var obj = new FindElements(root)
 * var param_1 = obj.find(target)
 */

File: leetcode-questions/1261-find-elements-in-a-contaminated-binary-tree/1261-find-elements-in-a-contaminated-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class FindElements:

    def __init__(self, root: Optional[TreeNode]):
        self.nodes = set([0])

        def traverse(root):
            if root.left:
                root.left.val = 2 * root.val + 1
                self.nodes.add(root.left.val)
                traverse(root.left)
            if root.right:
                root.right.val = 2 * root.val + 2
                self.nodes.add(root.right.val)
                traverse(root.right)

        root.val = 0
        traverse(root)

    def find(self, target: int) -> bool:
        return target in self.nodes


# Your FindElements object will be instantiated and called as such:
# obj = FindElements(root)
# param_1 = obj.find(target)
